WooCommerce/Classes/ViewRelated/Products/ProductsViewController.swift

```diff
@@ -1012,26 +1035,61 @@ private extension ProductsViewController {
     }
 }
 
+// MARK: - Favorite products
+//
+private extension ProductsViewController {
+    func observeFavoriteProductIDChanges() {
+        userDefaults.publisher(for: \.favoriteProductIDs)
```

Comment:
Nit: we can create a publisher from the use case and access it through the view model without revealing the implementation detail with user defaults here.